doc/pypit_file.rst

```diff
@@ -0,0 +1,249 @@
+
+.. _pypfile_by_line:
+
+Line by line
+============
+
+This section will instruct you on how to build a .pypit
+PYPIT file from scratch.  This is **not** recommended.
+The following documentation is mainly for guiding
+modifications to an existing PYPIT file.
+
+Naming
+++++++
+
+Create a .pypit file. Name it anything you want, but for example,
+it's useful to have: the instrument name, the grating or grism used,
+the dichroic, etc. For example, we could call our PYPIT file
+'lris_blue_long_600_4000_d560.pypit', for our data was collected
+on LRIS's blue arm, in long slit mode, using the 600/4000 grism
+and d560 dichroic.
+
+You can make any comments in your PYPIT file with a
+pound sign::
+
+    # This is a comment line
+
+We *recommend* you separate the main blocks of the .pypit file
+with comments.
+
+.. _run_block:
+
+Run block
++++++++++
+
+The first thing to include are changes to the
+default settings related to running PYPIT.
+The only one required is to set the name of the
+spectrograph::
+
+    run spectograph name_of_your_spectrograph
+
+We do recommend including several others, and the
+.pypit files made by the :ref:`pypit_pypfiles` script
+includes most of the following.
+Here are ones that one typically sets::
+
+    # Change the default settings
+    run ncpus 1                     # number of CPUs to use; can also negative integers,
+                                    so -1 means all but one CPU
+    run spectrograph lris_blue      # the spectrograph (+arm, if necessary) this set of data is from;
+                                    see README for list of available instruments
+    output verbosity 2                   # level of screen output; 0 = no output, 1 = low level of output;
+                                    2 = output everything
+    output overwrite True              # overwrite any existing output files?
+    output sorted lris_blue_long_600_4000_d560     # name of output files
+
+.. _reduce_block:
+
+Reduce block
+++++++++++++
+
+.. _data_block:
+
+Data block
+++++++++++
+
+Next, tell PYPIT where your raw data lives!
+One specifies the full path and may use wild cards
+to include a set of files.  If the data are compressed,
+include that extension.  Multiple entries are allowed
+
+Here is an example::
+
+    # Read in the data
+    data read
+     /Users/path/to/your/raw/data/*.fits
+    data end
+
+If you wish to skip individual files, you can specify these
+without the complete path, e.g.::
+
+    skip LB.20160406.17832.fits
+
+These will not be ignored as if they didn't exist.
+
+.. _spect_block:
+
+Spect block
++++++++++++
+
+Then, give PYPIT some information about your raw data. For
+example, PYPIT only accepts calibration files if they were
+created within a time window of the science frame of interest.
+You can set your own time window here. PYPIT also requires a
+certain number of each type of calibration file to be matched
+with the science frame, and here you can set what you want the
+minimum to be::
+
```

Comment:
Just curious: why is it suggested in the PR that "Going forward, the standard user may be advised to avoid the spec read/end block, aside from setting arc number 1", but there's also a few new sections in the docs referring to editing the spect block?